### PR TITLE
Add cloud sql instance flags

### PR DESCRIPTION
**Pull request by adityanatraj**
# why

there was an issue that i got notified of about adding
support for the set of cloud sql instance command line flags [here](https://github.com/oliver006/drone-cloud-run/issues/3).

notes:
- since you can only set one i used a notation that i describe in 
the readme. i dunno if thats good or bad. i felt it was nice because
it encodes the "only use one flag" rather easily.

# what
this pr:
- adds support for the cloud sql instances changes

### CODE REVIEW

**Comment by oliver006** [comment]
Thanks for the PR!

**Comment by oliver006** [comment]
Odd, this didn't trigger a drone run with the tests, let me check what's going on.

File: main_test.go [hunk 1]
```diff
@@ -132,6 +132,7 @@ func TestParseAndRunConfig(t *testing.T) {
 				"PLUGIN_MEMORY":                "128Mi",
 				"PLUGIN_TIMEOUT":               "10s",
 				"PLUGIN_REGION":                "us-central1",
+				"PLUGIN_CLOUD_SQL_UPDATE":      "-my-proj:east2:db2",
```
Comment (oliver006) [inline main_test.go:135]: Can you also add a tests the error case? E.g. with an invalid CloudSQL string with `first char != {+-=c}` ?

**Review by oliver006 — COMMENTED** [review]
I'm fine with the change but a general idea: how about rather adding an all-purpose way to add any sort of addtl parameters to the gcloud command to make this future proof without having to add a new parameters each time.

File: main_test.go (hunk 1 above)
Comment (adityanatraj) [reply main_test.go:135]: :+1:

**Comment by adityanatraj** [comment]
yeah, agreed; it'd certainly make this more flexible without manual updates.

it also brings up a related point about this feature too. the way i've implemented it, this drone plugin has an implicit safety check around using ONLY 1 of the cloud sql changing flags. in a more generic (and perhaps naive) version, it might just pass-through the direct flags and pass that error checking onto the `gcloud` command, which maybe i should have done here instead of this made up notation...

**Comment by oliver006** [comment]
I don't have a strong preference either way, let me know if you rather want to keep this as is or change it.

File: README.md [hunk 2]
```diff
@@ -21,6 +21,7 @@ steps:
       memory: 512Mi
       region: us-central1
       allow_unauthenticated: true
+      cloud_sql_instances: +instance1,instance2
```
Comment (oliver006) [inline README.md:24]: This needs to be "cloud_sql_update", no?

**Comment by adityanatraj** [comment]
rfr.

i updated this pr to go with a more generalized "additional flags" (`addl_flags`) that can be arbitrary key values which are the exact flags from the gcloud documentation but without the prefix dashes (because possible YAML confusion).

File: main.go [hunk 3]
```diff
@@ -88,6 +90,13 @@ func parseConfig() (*Config, error) {
 		log.Printf("os.Getenv(PLUGIN_ENVIRONMENT): %s", envStr)
 	}
 
+	addlFlagsStr := os.Getenv("PLUGIN_ADDL_FLAGS")
+	if err := json.Unmarshal([]byte(addlFlagsStr), &cfg.AdditionalFlags); err != nil && addlFlagsStr != "" {
+		log.Printf("json.Unmarshal() err: %s", err)
+		log.Printf("os.Getenv(PLUGIN_ADDL_FLAGS): %s", envStr)
```
Comment (oliver006) [inline main.go:96]: envStr --> addlFlagsStr ?

**Review by oliver006 — COMMENTED** [review]
I like it - nice work!
One minor thing to fix and we can ship this.

File: main.go (hunk 3 above)
Comment (adityanatraj) [reply main.go]: 🤦

**Review by oliver006 — APPROVED** [review]

**Pull request merged**